client/my-sites/stats/wordads/index.jsx

```diff
@@ -0,0 +1,194 @@
+import Main from 'components/main';
+import StatsNavigation from 'blocks/stats-navigation';
+import StatsPeriodNavigation from '../stats-period-navigation';
+import DatePicker from '../stats-date-picker';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+import WordAdsChartTabs from '../wordads-chart-tabs';
+import titlecase from 'to-title-case';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import JetpackColophon from 'components/jetpack-colophon';
+import WordAdsEarnings from './earnings';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { recordGoogleEvent } from 'state/analytics/actions';
+import PrivacyPolicyBanner from 'blocks/privacy-policy-banner';
+import StickyPanel from 'components/sticky-panel';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class WordAds extends Component {
+	state = {
+		chartTab: this.props.chartTab,
+		tabSwitched: false,
+	};
+
+	UNSAFE_componentWillReceiveProps( nextProps ) {
+		// @TODO update
+		if ( ! this.state.tabSwitched && this.state.chartTab !== nextProps.chartTab ) {
+			this.setState( {
+				tabSwitched: true,
+				chartTab: nextProps.chartTab,
+			} );
+		}
+	}
+
+	barClick = bar => {
+		this.props.recordGoogleEvent( 'Stats', 'Clicked Chart Bar' );
+		page.redirect( this.props.path + '?startDate=' + bar.data.period );
+	};
+
+	switchChart = tab => {
+		if ( ! tab.loading && tab.attr !== this.state.chartTab ) {
+			this.props.recordGoogleEvent( 'Stats', 'Clicked ' + titlecase( tab.attr ) + ' Tab' );
+			this.setState( {
+				chartTab: tab.attr,
+				tabSwitched: true,
+			} );
+		}
+	};
+
+	formatCurrency = value => {
+		return '$' + numberFormat( value, 2 );
+	};
+
+	render() {
+		const { date, site, siteId, slug, translate } = this.props;
+
+		const charts = [
+			{
+				attr: 'impressions',
+				legendOptions: [ 'impressions' ],
+				gridicon: 'visible',
+				label: translate( 'Ads Served' ),
+			},
+			{
+				attr: 'cpm',
+				gridicon: 'stats-alt',
+				label: translate( 'Average CPM' ),
+				format: this.formatCurrency,
+			},
+			{
+				attr: 'revenue',
+				gridicon: 'money',
+				label: translate( 'Revenue' ),
+				format: this.formatCurrency,
+			},
+		];
+
+		const { period } = this.props.period;
+
+		const todayObject = new Date();
+		const yesterdayObject = new Date();
+		yesterdayObject.setDate( yesterdayObject.getDate() - 1 );
+
+		const today =
+			( '0' + todayObject.getFullYear() ).slice( -4 ) +
+			'-' +
+			( '0' + ( todayObject.getMonth() + 1 ) ).slice( -2 ) +
+			'-' +
+			( '0' + todayObject.getDate() ).slice( -2 );
+
+		const yesterday =
+			( '0' + yesterdayObject.getFullYear() ).slice( -4 ) +
+			'-' +
+			( '0' + ( yesterdayObject.getMonth() + 1 ) ).slice( -2 ) +
+			'-' +
+			( '0' + yesterdayObject.getDate() ).slice( -2 );
+
+		const queryDate =
+			today === date.format( 'YYYY-MM-DD' ) ? yesterday : date.format( 'YYYY-MM-DD' );
```

Comment:
What does this code do? If asking for stats for today, show me yesterday instead? It's hard to understand for me without any explanation.